discord/invite.py

```diff
@@ -315,9 +323,14 @@ def __init__(self, *, state, data: InvitePayload):
         inviter_data = data.get('inviter')
         self.inviter = None if inviter_data is None else self._state.store_user(inviter_data)
         self.channel = data.get('channel')
-        target_user_data = data.get('target_user')
-        self.target_user = None if target_user_data is None else self._state.store_user(target_user_data)
-        self.target_type = try_enum(InviteTarget, data.get('target_type', 0))
+
+        self.target_type = try_enum(InviteTarget, data.get("target_type", 0))
+
+        target_user = data.get('target_user')
+        self.target_user = state.store_user(target_user) if target_user else None
+
+        application = data.get("target_application")
+        self.target_application = AppInfo(data=application, state=state) if application else None
```

Comment:
This entire diff needs work since it's needlessly overwriting someone's previous work. Also, single quotes not double quotes.